jkube-kit/enricher/api/src/test/java/org/eclipse/jkube/kit/enricher/api/util/KubernetesResourceUtilTest.java

```diff
@@ -300,6 +304,20 @@ public void getNameWithSuffix_withUnknownMapping_shouldReturnCR() {
         assertEquals("name-cr", getNameWithSuffix("name", "VeryCustomKind"));
     }
 
+    @Test
+    public void updateKindFilenameMappings_whenAddsCronTabMapping_updatesKindToFileNameMapper() {
```

Comment:
Add a setup step (`@BeforeEach`) to clear the mappings

Reply:
When I add `clear()` call in `@Before`, other tests start failing

Reply:
We've got a problem here then

Reply:
I managed to do this by taking backup of these maps in `@BeforeAll` and then clear + putAll in `@Before`

Reply:
clear the mappings and run `initializeKindFilenameMapper();` which should be accessible from the Test class

Reply:
The mutability of this field is really bad. We want to make sure we **always** start with a clean consistent slate.
Saving and restoring the previous state is a good workaround, but still doesn't guarantee that the same input is always provided to this test.
If we can reinitialize the mappings for every execution, it's a better approach.